included removing sparse values from dictionnary

0xrushi opened the pull request:
@MrDupin please have a look at this

0xrushi committed:
included removing sparse values from dictionnary

MrDupin commented on learning.py:
```diff
@@ -298,6 +298,18 @@ def sample(self):
         return self.sampler()
+        
+    def remove_sparse(self,sparse_value):
+    	""" keep items with probablity sparse value 
```
Capitalize comment start. Also, it should be 'Keep items with probability greater than sparse value' or something like that.

MrDupin commented on learning.py:
```diff
@@ -298,6 +298,18 @@ def sample(self):
                                             list(self.dictionary.values()))
         return self.sampler()
+        
+    def remove_sparse(self,sparse_value):
```
Add space after comma.

MrDupin commented on learning.py:
```diff
@@ -298,6 +298,18 @@ def sample(self):
+        
+    def remove_sparse(self,sparse_value):
+    	""" keep items with probablity sparse value 
+    	and remove all others , eg if sparse_value=0.98 items with probablity
```
Remove space before comma and add period after 'eg'.

MrDupin commented on learning.py:
```diff
@@ -298,6 +298,18 @@ def sample(self):
+    def remove_sparse(self,sparse_value):
+    	""" keep items with probablity sparse value 
+    	and remove all others , eg if sparse_value=0.98 items with probablity
+    	0.02 would be removed"""
```
You are not only removing items with probability 0.02, you are removing items with probability less than 0.02. Also, add a period at the end of the comment.

antmarakis reviewed (commented):
I am not sure where or how this will be used, so I am not sure if we need this. Can you provide an example of this? Generally we try to only add code that has some use (either educational or implemention-wise). If this simply for educational purposes, you should add a bit of a description.

0xrushi committed:
modified comments for remove_sparse()

@MrDupin sorry, i had seen its real implementation somewhere but i'm not able to remember. There was an built in class for it in some library. Will update here if i find something.

dmeoli commented:
Thanks @0xrushi. This remove_sparse method deletes from self.dictionary while iterating over self.dictionary.items(), which raises 'dictionary changed size during iteration', and uses tab indentation. Closing; a fixed, PEP 8 version (iterate over a list copy of the keys) in a focused PR would be welcome. Thanks!
